plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -913,16 +944,17 @@ void TransformationMatrix::initialiseToPrim(bool readFromPrim, Transform* transf
         m_flags |= kPrimHasPivot;
         if(readFromPrim)
         {
-          internal_readPoint(m_scalePivotFromUsd, op);
-          m_rotatePivotFromUsd = m_scalePivotFromUsd;
+          MPoint ptFromUsd;
+          internal_readPoint(ptFromUsd, op);
           if(transformNode)
```

Comment:
I'm happy to add tweak zeroing to all the cases - this is one of those things that seems important that I didn't know because I don't have a deep understanding of the code I'm changing.
The reason I started using the temp variables everywhere is I noticed, as you said, callbacks being triggered at inopportune times with the result of undesirable USD "over" settings of values being added that covered up existing animation with over-ed constant values.  A number of the code changes in this pull request are to try and prevent callbacks from triggering too many times or at times that cause problems.
(Note that I also noticed an obviously wrong tweak value being auto-computed and set a few times while I was trying to figure out the code, so there may be some problem hiding in the tweak mechanism.  Unfortunately I don't have a reproducible case of this for you.)

Reply:
Hmm, do you know how / why using the temp variables prevents overs being created?

Reply:
Added tweak zeroing to all the cases.
Regarding why using the temp variables prevents overs being created, my guess was that when the internal_read* methods were reading into the non-temp variables, they were triggering some callback because the object was actually being changed.  For some reason, using MPlug.setValue to update the object later wasn't having the same effect.  Not sure why, but I was printing out messages in callbacks to watch what was being triggered when and noticed behavior differences.  It's been a while so I don't remember all the details.

Reply:
When I first introduced the temp variables (back in the old AL repo days) the issue was that TransformationMatrix::pushToPrim() was looping over all the transform operators, pushing those values to the primitive and updating the m_* variables (and they would lose their values). For instance, when the translateX plug was being set, TransformationMatrix::pushToPrim() would be called, then MPxTransformationMatrix::translationValue would be pushed to the primitive and m_translationFromUsd updated with that value. However, at that moment MPxTransformationMatrix::translationValue only has the value read for the X component of the translation - Y and Z have not been set yet. The values read for Y and Z got lost

Reply:
Ok - so that definitely explains why this would be necessary if pushToPrim was still enabled.

However - I think that now that pushToPrim is disabled during this function, I think this shouldn't be necessary - and the simpler "write directly to m_scalePivotFromUsd" method should work.

Of course, knowing things like this "for sure" is precisely why I'd like to see tests for this!

Reply:
@pilarmolinalopez - Now that you mention it, I'd also run intothe same strange problem!  Only one of the xyz components of a vector was correctly updated at some point, so when it came time for the code to check if the saved vector was the same as the value we were trying to set it to, the answer came back that no it wasn't the same.  This resulted in some operation being performed that wouldn't have been performed if the x y and z had all been properly updated at the same time.